protocol/binary/reader_benchmark_test.go

```diff
@@ -0,0 +1,48 @@
+package binary
+
+import (
+	"bytes"
+	"io"
+	"testing"
+
+	"go.uber.org/thriftrw/wire"
+)
+
+func Benchmark_readStruct(b *testing.B) {
+	for _, bench := range []struct {
+		name      string
+		numFields int
+	}{
+		{
+			name:      "small struct with 64 fields",
+			numFields: 64,
+		},
+		{
+			name:      "large struct with 512 fields",
+			numFields: 512,
+		},
+	} {
+		reader := NewReader(constructThriftStruct(bench.numFields))
+
+		b.Run(bench.name, func(b *testing.B) {
+			for i := 0; i < b.N; i++ {
```

Comment:
When using subtests/subbenchmarks, setup specific to that benchmark/test should normally be inside the new function.

You can call b.ResetTimer to not have `NewReader(..)` count against the performance:
```suggestion
		b.Run(bench.name, func(b *testing.B) {
			reader := NewReader(constructThriftStruct(bench.numFields))
			b.ResetTImer()
			for i := 0; i < b.N; i++ {
```